cve_bin_tool/NVDAutoUpdate.py

```diff
@@ -283,3 +283,63 @@ def get_cves(self, *vendor_product_pairs):
     def get_cvelist_if_stale(self):
         """ Update CVEs data from NVD if stale."""
         get_cvelist_if_stale(self.nvddir, self.dbname)
+
+class MappingDb:
+    def __init__(self, table_name, mapping_function, duration):
+        self.table_name = table_name
+        self.mapping_function = mapping_function
+        self.disk_location = DISK_LOCATION_DEFAULT
+        self.duration = duration
+        self.conn = None
+        self.cursor = None
+
+    @property
+    def dbname(self):
+        return os.path.join(self.disk_location, 'version_map.db')
+
+    def open(self):
+        self.conn = sqlite3.connect(self.dbname)
+        self.cursor = self.conn.cursor()
+
+    def close(self):
+        self.cursor.close()
+        self.conn.close()
+        self.conn = None
+        self.cursor = None
+
+    def __enter__(self):
+        self.open()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.close()
+
+    def get_mapping_data(self):
```

Comment:
Please add a docstring here.  I haven't turned on pylint in CI yet but I'd like to do it soon (hopefully next week if my whole family stops being sick), so we might as well future-proof now.